test_runner/fixtures/neon_fixtures.py

```diff
@@ -324,14 +323,43 @@ async def connect_async(self, **kwargs) -> asyncpg.Connection:
         # Convert options='-c<key>=<val>' to server_settings
         if 'options' in conn_options:
             options = conn_options.pop('options')
-            for match in re.finditer('-c(\w*)=(\w*)', options):
-                key = match.group(1)
-                val = match.group(2)
-                if 'server_options' in conn_options:
-                    conn_options['server_settings'].update({key: val})
+            for match in options.split(" "):
+                print("match", match)
+                if "-c" in match:
+                    matches = [x for x in re.finditer('-c(\w*)=(\w*)', match)]
+                    assert len(matches) == 1
+                    match = matches[0]
+                    key = match.group(1)
+                    val = match.group(2)
+                    if 'server_settings' in conn_options:
+                        conn_options['server_settings'].update({key: val})
+                    else:
+                        conn_options['server_settings'] = {key: val}
+                elif "=" in match:
+                    parts = match.split("=")
+                    assert len(parts) == 2
+                    key, val = parts[0], parts[1]
+                    if 'server_settings' in conn_options:
+                        conn_options['server_settings'].update({key: val})
+                    else:
+                        conn_options['server_settings'] = {key: val}
                 else:
-                    conn_options['server_settings'] = {key: val}
-        return await asyncpg.connect(**conn_options)
+                    # invalid options
+                    assert False
+        return conn_options
+
+    async def connect_async(self, callback = None, **kwargs) -> asyncpg.Connection:
+        """
+        Connect to the node from async python.
+        Returns asyncpg's connection object.
+        """
+
+        conn_options = self.connect_async_prepare_connection_options(**kwargs)
+
+        if callback is None:
+            return await asyncpg.connect(**conn_options)
+        else:
+            return (await asyncpg.connect(**conn_options)).add_log_listener(callback)
```

Comment:
When running the test, it stalls here. Most probably because the proxy is actually waiting on the authentication step, which will never happen because it's stalling the rest of the python code. The callback is never triggered. Any suggestions?

Reply:
this is called from test_psql_session_id_second_attempt.